scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd

```diff
@@ -0,0 +1,410 @@
+- `CD14`
+  - "CD14 is a glycolipid-anchored membrane glycoprotein expressed on cells of the myelomonocyte lineage including **monocytes**, **macrophages**, and some **granulocytes**." [source](https://www.sciencedirect.com/topics/neuroscience/cd14)
+- `CD16`
+  - "CD16 is involved in antibody-dependent cell-mediated cytotoxicity and is expressed on large granular lymphocytes (LGLs) of both **NK- and T-cell types**." [source](https://www.sciencedirect.com/topics/neuroscience/cd16)
+- `CD19`
+  - "CD19 is an IgSF surface glycoprotein of 95 kDa that is expressed from the earliest stages of **B cell** development until plasma cell terminal differentiation, when its expression is lost." [source](https://www.sciencedirect.com/topics/neuroscience/cd19)
+- `CD56`
+  - "CD56 expression is also found in, among others, the hematopoietic system. Here, the expression of CD56 is mostly associated with, but not limited to, **natural killer cells**. CD56 has been detected on other lymphoid cells, including **gamma delta Τ cells and activated CD8+ T cells**, as well as on **dendritic cells**." [source](https://en.wikipedia.org/wiki/Neural_cell_adhesion_molecule)
+
+
+
+Let's color cells by their most prevalent ADT:
+
+```{r does anybody have a UMAP}
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "max_adt", 
+                       point_alpha = 0.6, point_size = 0.25)
+```
+
+
+
+
+## Cell type annotation from RNA
+
+We can introduce annotation with `SingleR`, and it might be a good idea to show how the results are highly influenced by the reference. 
+For now, I'm just using the `HumanPrimaryCellAtlasData` reference as a starting place.
+Note that all references here involve some caching, and we can expect some interactive prompting to go by for setting up these caches that we'll have to walk everyone through.
+
+
+```{r}
+ref_data <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE) # pick one for now to start
+preds <- SingleR::SingleR(
+  test = norm_sce, 
+  ref = ref_data,
+  labels = ref_data$label.main, # "coarse-grained" labels
+  # faster!
+  BPPARAM = BiocParallel::MulticoreParam(4)
+)
+preds
+
+# We can place these into the sce
+norm_sce$celltype_predictions <- preds$pruned.labels
+
+# **heavily** biased towards monocytes and T-cells
+celltype_counts <- preds %>%
+  as.data.frame() %>%
+  dplyr::count(pruned.labels) %>%
+  dplyr::arrange(desc(n))
+
+celltype_counts
+
+```
+
+
+
+Visualize the predictions in a couple ways:
+```{r}
+SingleR::plotScoreHeatmap(preds)
+```
+
+
+```{r UMAP for the people}
+# Not too shabby!
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "celltype_predictions", 
+                       point_alpha = 0.4, point_size = 0.25)
+```
+
+
+It might be helpful to hone in on the most prevalent cell types only for ease of visualization, but this is a bit more advanced to dive into.
```

Comment:
I actually think it might be worth showing both plots with all the types and then the plot with just the top cell types. Also offers an opportunity for talking about making good visualizations.